cmd/dagger/session.go

```diff
@@ -2,52 +2,118 @@ package main
 
 import (
 	"context"
+	"crypto/sha256"
 	"encoding/json"
 	"errors"
+	"fmt"
 	"io"
 	"net"
 	"net/http"
 	"os"
 	"os/signal"
+	"strings"
 	"syscall"
 	"time"
 
 	"github.com/dagger/dagger/engine"
 	internalengine "github.com/dagger/dagger/internal/engine"
 	"github.com/dagger/dagger/router"
+	"github.com/go-git/go-git/v5"
 	"github.com/google/uuid"
 	"github.com/spf13/cobra"
 )
 
+var sessionLabels labels
+
 func sessionCmd() *cobra.Command {
-	return &cobra.Command{
+	cmd := &cobra.Command{
 		Use:          "session",
 		Long:         "WARNING: this is an internal-only command used by Dagger SDKs to communicate with the Dagger engine. It is not intended to be used by humans directly.",
 		Hidden:       true,
 		RunE:         EngineSession,
 		SilenceUsage: true,
 	}
+	cmd.Flags().Var(&sessionLabels, "label", "label that identifies the source of this session (e.g, --label 'sdk:python' --label 'sdk_version:0.5.2' --label 'sdk_async:true')")
+	return cmd
 }
 
 type connectParams struct {
 	Port         int    `json:"port"`
 	SessionToken string `json:"session_token"`
 }
 
+func isCI() bool {
+	return os.Getenv("CI") != "" || // GitHub Actions, Travis CI, CircleCI, Cirrus CI, GitLab CI, AppVeyor, CodeShip, dsari
+		os.Getenv("BUILD_NUMBER") != "" || // Jenkins, TeamCity
+		os.Getenv("RUN_ID") != "" // TaskCluster, dsari
+}
+
+func getGitInfo() (string, string, error) {
+	repo, err := git.PlainOpen(".")
+	if err != nil {
+		return "", "", err
+	}
+
+	config, err := repo.Config()
+	if err != nil {
+		return "", "", err
+	}
+
+	committerEmail := config.User.Email
+	committerHash := fmt.Sprintf("%x", sha256.Sum256([]byte(committerEmail)))
+
+	remote, err := repo.Remote("origin")
+	var repoURL string
```

Comment:
I think we can remove this, wdyt ?

Reply:
I'm sorry if I expressed myself poorly.

I wanted to point `var repoURL string`